### PR TITLE
add sensitive info flag to cli

### PR DESCRIPTION
**Pull request by ryanfkeepers**
The sensitive-info flag allows CLI users to specify the Concealer algorithm used to handle PII and
other sensitive info in logging.  Default alg is 'hash', which produces a truncated hmacsha256 string.
'mask' for flat '***' replacement and 'plaintext'
are also available.

SDK consumer are expected to configure clues
themselves, rather than specifying a config option.

---

#### Does this PR need a docs update or release note?

- [x] :no_entry: No

#### Type of change

- [x] :sunflower: Feature

#### Issue(s)

* #2024

#### Test Plan

- [x] :muscle: Manual
- [x] :zap: Unit test

### CODE REVIEW

**Comment by ryanfkeepers** [comment]
Logging examples:

`--sensitive-info hash` (aka, default)
```
2023-03-31T11:26:04.837-0600	INFO	observe/observe.go:191	Connecting to repository
{"acct_id": "f46d7a30b7282c86"}
```

`--sensitive-info mask`
```
2023-03-31T11:25:12.202-0600	INFO	observe/observe.go:191	Connecting to repository
{"acct_id": "***"}
```

`--sensitive-info plaintext`
```
2023-03-31T11:25:39.482-0600	INFO	observe/observe.go:191	Connecting to repository
{"acct_id": "deadbeef-e94c-4ea5-8a8a-0000beefdead"}
```

**Comment by ryanfkeepers** [comment]
@gmatev Pinging you for a review on flag naming and explanation:

current help text (the flag is hidden until we're sure pii handling is covered)
```
      --sensitive-info string   set the format for sensitive info in logs to hash|mask|plaintext (default "hash")
```

also, preferred default?

File: src/pkg/logger/logger.go [hunk 1]
```diff
@@ -65,58 +78,102 @@ func defaultLogLocation() string {
...
+		ReadableLogsFN,
+		false,
 		"minimizes log output for console readability: removes the file and date, colors the level")
-	//nolint:errcheck
-	fs.MarkHidden(readableLogsFN)
+
+	fs.StringVar(
+		&SensitiveInfoFV,
+		SensitiveInfoFN,
+		PIIHash,
```
Comment (vkamra) [inline src/pkg/logger/logger.go:112]: @ryanfkeepers - let's set the default to `PIIPlainText` for CLI. Let users change this if needed.

**Review by vkamra — DISMISSED** [review]
Change requested to default value

File: src/pkg/logger/logger.go (hunk 1 above)
Comment (ryanfkeepers) [reply src/pkg/logger/logger.go]: Done

**Review by pandeyabs — APPROVED** [review]

**Pull request merged**